From: Lance Yang <ioworker0@gmail.com>
Subject: hung_task-add-detect-count-for-hung-tasks-fix
Date: Fri, 1 Nov 2024 19:48:33 +0800

sysctl_hung_task_detect_count is defined as an unsigned long, so we need
to use proc_doulongvec_minmax instead of proc_dointvec to handle it
correctly.

Link: https://lkml.kernel.org/r/20241101114833.8377-1-ioworker0@gmail.com
Signed-off-by: Lance Yang <ioworker0@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/hung_task.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/hung_task.c~hung_task-add-detect-count-for-hung-tasks-fix
+++ a/kernel/hung_task.c
@@ -330,7 +330,7 @@ static struct ctl_table hung_task_sysctl
 		.data		= &sysctl_hung_task_detect_count,
 		.maxlen		= sizeof(unsigned long),
 		.mode		= 0444,
-		.proc_handler	= proc_dointvec,
+		.proc_handler	= proc_doulongvec_minmax,
 	},
 };
 
_
